insights/parsers/vdsm_log.py

```diff
@@ -6,81 +6,155 @@
-        * **logname** - the logger's name
-        * **funcname** - the function name reporting the message
-        * **message** - the body of the message
-        * **raw_message** - the original unparsed line
-
-        The **datetime** field is only present if the `timestamp` field
-        is present contains a time stamp in the valid format.
-
-        If the line is too short, for some reason, then as many fields as
-        possible are pulled from the line.
-
-    Example:
+    Usage:
+        >>> from insights.parsers.vdsm_log import VDSMLog
+        >>> from insights.tests import context_wrap
+        >>> shared = {VDSMLog: VDSMLog(context_wrap(VDSM_LOG))}
         >>> log = shared[VDSMLog]
-        >>> log.get('TaskManager')[0]
-        {'raw_message': 'Thread-4662::DEBUG::2015-05-08 18:01:05,560::task::595::TaskManager.Task::(_updateState) Task=`9a7948f6-b6d9-42c2-b91f-7e0346dfc1d6`::moving from state init -> state preparing',
-         'thread': 'Thread-4662',
+        >>> log.get('TaskManager')
+        'Thread-4662::DEBUG::2015-05-08 18:01:05,560::task::595::TaskManager.Task::(_updateState) Task=`9a7948f6-b6d9-42c2-b91f-7e0346dfc1d6`::moving from state init -> state preparing'
+        >>> list(log.parse_lines(log.get('TaskManager')))[0]  # from generator to list to subscript
+        {'thread': 'Thread-4662',
          'level': 'DEBUG',
-         'timestamp': '2015-05-08 18:01:05,560',
-         'datetime': datetime(2015, 5, 8, 18, 1, 5),
+         'asctime': datetime(2015, 5, 8, 18, 1, 5, 56000),
          'module': 'task',
          'line': '595',
          'logname': 'TaskManager.Task',
-         'funcname': '_updateState',
          'message': 'Task=`9a7948f6-b6d9-42c2-b91f-7e0346dfc1d6`::moving from state init -> state preparing'
         }
+
+
+    Note: VDSM version 4 has different logs format than version 3.
+    VDSM version 4 logs parser is designed to closely match log format
+    as referred in ``/etc/vdsm/logger.conf`` which is as below::
+
+        format: %(asctime)s %(levelname)-5s (%(threadName)s) [%(name)s] %(message)s (%(module)s:%(lineno)d)
+
+    Sample logs from VDSM version 4::
+
+        2017-04-18 14:00:00,000+0200 INFO  (jsonrpc/2) [jsonrpc.JsonRpcServer] RPC call Host.getStats succeeded in 0.02 seconds (__init__:515)
+        2017-04-18 14:00:01,807+0200 INFO  (Reactor thread) [ProtocolDetector.AcceptorImpl] Accepted connection from ::ffff:10.34.60.219:49213 (protocoldetector:72)
+        2017-04-18 14:00:01,808+0200 ERROR (Reactor thread) [ProtocolDetector.SSLHandshakeDispatcher] Error during handshake: unexpected eof (m2cutils:304)
+        2017-04-18 14:00:03,304+0200 INFO  (jsonrpc/6) [jsonrpc.JsonRpcServer] RPC call Host.getAllVmStats succeeded in 0.00 seconds (__init__:515)
+        2017-04-18 14:00:05,870+0200 INFO  (jsonrpc/7) [dispatcher] Run and protect: getSpmStatus(spUUID=u'00000002-0002-0002-0002-00000000024f', options=None) (logUtils:51)
+
+    Usage:
```

Comment:
Same comment as above.

Reply:
@bfahr Thanks for pointing out that. I reverted 'Usage' back to 'Example'